neutrons/SNAPRed
# Ewm8253 normcal index being misread

## Description of work

Requires this data change to be merged to pass integration tests:
https://code.ornl.gov/sns-hfir-scse/infrastructure/test-data/snapred-data/-/merge_requests/1

## Explanation of work

It was found that the index was not appropriately failing when a version for a specified run was not found.
This was in part due to its subbing in the "current" version if it received `None` as an input, which it also happened to return when it couldnt find a version.  This then created a scenario in which not finding a record resulted in pulling the `current` version.

The `current` version being arbitrarily the last one done.

## To test

Run through the full range of workflows and see that they work.
Then mess with the calibration/normalization indexes and see how the reduction workflow responds.  Confirm that it matches up with what is expected.

## Link to EWM item


[EWM#8253](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=8253)

### Verification




### Acceptance Criteria



This list is for ease of reference, and does not replace reading the EWM story as part of the review.  Verify this list matches the EWM story before reviewing.

to replicate:
1. Set calibration home = `/SNS/SNAP/shared/Calibration_next`
2. Reduce run 57482 (in lite mode)
3. Data are reduced, using version 2 of normcal in the folder `/SNS/SNAP/shared/Calibration_next/Powder/04bd2c53f6bf6754/lite/normalization/`

However, if the normalizationIndex.json is inspected its content should mean that there is no normcal available for run 57482:

```
[
    {
        "runNumber": "58809",
        "useLiteMode": true,
        "appliesTo": ">=58882",
        "comments": "test of v1.1.0rc1",
        "author": "mg",
        "timestamp": 1729258319.2569995,
        "version": 1
    },
    {
        "runNumber": "58809",
        "useLiteMode": true,
        "appliesTo": ">=58809",
        "comments": " 3mm V-Nb sample",
        "author": "Malcolm",
        "timestamp": 1730229952.2204628,
        "version": 2
    }
]
```
The normcal is only available for run numbers >=58809.

- [x] The expected behaviour would be that SNAPRed gives a warning that normcal does not exist for run 57482 and proceeds to manage that situation.
